File: src/main/java/com/mygaienko/model/Work.java
package com.mygaienko.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;

/**
 * Created by enda1n on 12.11.2016.
 */
@Data
@Entity(name = "work")
@EqualsAndHashCode(of = "id")
public class Work implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String description;

    private BigDecimal price;

    @Enumerated(EnumType.STRING)
    private WorkStatus status;

    @ManyToOne
    @JoinColumn(name="request_id")
    private Request request;

    @ManyToOne
    @JoinColumn(name = "master_id")
    private User master;
}

File: src/main/webapp/app/_models/work.ts
import { UserInfo } from "./user-info";
export class Work {

    id: string;
    description: string;
    price: string;
    status: string;
    master: UserInfo;
    requestId: string


    constructor(description: string, price: string, status: string, master: UserInfo, requestId: string) {
        this.description = description;
        this.price = price;
        this.status = status;
        this.master = master;
        this.requestId = requestId;
    }
}